### PR TITLE
Make behavioural difference to Python set methods clear in C-API docs

### PR DESCRIPTION
**Pull request by scoder**
See https://github.com/capi-workgroup/problems/issues/6

----
:books: Documentation preview :books:: https://cpython-previews--104379.org.readthedocs.build/

### CODE REVIEW

File: Doc/c-api/set.rst [hunk 1]
```diff
@@ -121,12 +121,15 @@ or :class:`frozenset` or instances of their subtypes.
...
-   convert unhashable sets into temporary frozensets.  Raise a :exc:`TypeError` if
-   the *key* is unhashable. Raise :exc:`PyExc_SystemError` if *anyset* is not a
+   Return ``1`` if found, ``0`` if not found, and ``-1`` if an error is encountered.
+   Raise a :exc:`TypeError` if the *key* is unhashable.
+   Raise :exc:`PyExc_SystemError` if *anyset* is not a
    :class:`set`, :class:`frozenset`, or an instance of a subtype.
 
+   .. note::
+      Unlike the Python :meth:`__contains__` method, this function does not
+      automatically convert unhashable sets into temporary frozensets.
```
Comment (scoder) [inline Doc/c-api/set.rst]:
```suggestion
   .. note::
      Unlike the Python :meth:`__contains__` method, this function does not
      automatically convert unhashable sets into temporary frozensets.
```

**Comment by rhettinger** [comment]
It looks to me like all the information is already there.  I don't really see a point in making the docs less readable by highlighting the least important information.

**Comment by scoder** [comment]
> I don't really see a point in making the docs less readable by highlighting the least important information.

I agree that it's not relevant for many use cases, but it's a gotcha if it is. It definitely was a gotcha for me, Cython had this feature missing for quite a while, until someone noticed that it didn't work there (that was years ago). So – it's clearly not often needed, but that also means that it's very unlikely to be tested for and very likely to go unnoticed. And apparently, others have noticed the difference, too, and written up ticket https://github.com/capi-workgroup/problems/issues/6.

OTOH, you could argue that the function name is such an obvious equivalent to `set.__contains__` that people won't bother looking it up in the documentation at all, and, thus, won't notice the sentence either way, however visible it is in the docs…

**Comment by rhettinger** [comment]
Thanks for the suggestion, but I'm going to decline.   I don't buy that of the three sentences in the docs, that this one needs to be moved into a large box.  It reads plainly enough and it covers a really minor, mostly unused feature.

**Pull request closed without merging**